src/homedelivery/query.js

```diff
@@ -27,17 +27,36 @@ async function queryZuora (deliveryDate, config: Config) {
         SoldToContact.PostalCode,
         SoldToContact.State,
         SoldToContact.workPhone,
-        SoldToContact.SpecialDeliveryInstructions__c
-    FROM
+        SoldToContact.SpecialDeliveryInstructions__c,
+        Subscription.Version,
+        Subscription.Status,
+        RateplanCharge.Version,
+        RatePlanCharge.EffectiveEndDate,
+        RatePlan.AmendmentType,
+        RateplanCharge.Id
+        
+      FROM
       rateplancharge
     WHERE
      (Subscription.Status = 'Active' OR Subscription.Status = 'Cancelled') AND
      ProductRatePlanCharge.ProductType__c = 'Print ${deliveryDay}' AND
      Product.Name = 'Newspaper Delivery' AND
+     (RatePlan.AmendmentType IS NULL OR RatePlan.AmendmentType != 'RemoveProduct') AND
      RatePlanCharge.EffectiveStartDate <= '${formattedDate}' AND
-     (Subscription.Status = 'Active' AND RatePlanCharge.EffectiveEndDate >= '${formattedDate}' OR
-      Subscription.Status = 'Cancelled' AND RatePlanCharge.EffectiveEndDate > '${formattedDate}'
-     ) AND
+     (
+      ( 
+        Subscription.Status = 'Active' AND Subscription.AutoRenew = true AND RatePlanCharge.IsLastSegment = true
+      )
+      OR
+      (
+        Subscription.Status = 'Active' AND RatePlanCharge.EffectiveEndDate >= '${formattedDate}'
+      )
+      OR
+      ( 
+        Subscription.Status = 'Cancelled' AND RatePlanCharge.EffectiveEndDate > '${formattedDate}' 
+      )
+     ) 
+     AND
```

Comment:
we had to ask for the last segment to avoid getting old rate plan charges:
https://knowledgecenter.zuora.com/DC_Developers/G_SOAP_API/E1_SOAP_API_Object_Reference/RatePlanCharge#Segmented_rate_plan_charges

Maybe we should check that EffectiveStartDate <= Today < EffectiveEndDate in the rateplancharge
what do you think @paulbrown1982 ?